Updated init template with `.mise.toml` and `Package.swift` files

dxmvsh opened the pull request:
Resolves https://github.com/tuist/tuist/issues/5897

### Short description 📝

Add `.mise.toml` and `Tuist/Package.swift` generation in init command default template.

### How to test the changes locally 🧐
Create empty folder and run `tuist init`. This will generate iOS platform app with your folder name. These changes additionally create `.mise.toml` file with tuist version you are using and `Tuist/Package.swift` to add your external dependencies.

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint:fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

dxmvsh commented on Sources/TuistKit/Services/InitService.swift:
```diff
@@ -117,6 +117,7 @@ class InitService {
                 let parsedAttributes = try parseAttributes(
                     name: name,
                     platform: platform,
+                    tuistVersion: Constants.version,
```
Using user's current tuist version from `Constants.version` and I don't know if its okay. 
Is there any other way to get which tuist version is running at runtime?

pepicrft replied:
We've recently moved the version into the tuist-cloud part of the CLI to ease our release process, and therefore that `Constants.version` is no longer used. What I recommend doing is the following:

```bash
let result = try System.shared.capture([CommandLine.arguments[0], "version"])
```

`CommandLine.arguments[0]` should be the path to Tuist itself, which is usually the first argument of the process being executed, and then you call `version` on it.

Long-term we'll refactor the code so that the implementation is not that meta (tuist calling tuist)

danieleformichelli commented:
@allcontributors add @dxmvsh for code

pepicrft commented on Templates/default/Package.stencil:
```diff
@@ -0,0 +1,21 @@
+    name: "{{ name }}",
+    dependencies: [
+        // Add your own dependencies here:
+        // .package(url: "https://github.com/Alamofire/Alamofire", from: "5.0.0"),
```
I'd also add:
```swift
// You can read more about dependencies here: https://docs.tuist.io/documentation/tuist/dependencies
```

dxmvsh commented:
Sorry for any inconvenience, will resolve all problems in next 2 days 😉

dxmvsh commented:
@pepicrft yes, I've looked through them, sorry for inconvenience. 

Currently the main tests that are failing are acceptance tests. As I changed to fetching tuist version via `tuist version` command, when tests are executed this command is not responding. I tried mocking `System.shared` instance, but I think that is not the correct way, because acceptance tests are depending on `System()` instance. 

I guess the only path here is to inject `Systeming` into `InitService`, and I thought that it is not the right way either. 

Do you think if I added new class instance that can get actual tuist version with `Systeming` instance injected into it, it would be okay? Or any other ideas?

dxmvsh commented:
@pepicrft 

I got updates. When tests are being executed `CommandLine.arguments[0]` is equal to `xctest` and it runs `xctest version` when trying to get current tuist version. So I decided to not depend on `CommandLine.arguments[0]` variable and directly added `tuist` in `try System.shared.capture(["tuist", "version"])`.

As unit tests for `InitService` highly depend on what `tuist version` command executes, I added new class named `TuistVersionLoader` in order to separate probably often used logic, add injection of `Systeming` instance and etc. 

Hope it is fine and I will be waiting for your feedback!

fortmarek reviewed (approved):
Amazing work 👏
